aptos-move/aptos-native-interface/src/helpers.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/// Like `pop_arg!` but for safe natives that return `SafeNativeResult<T>`. Will return a
+/// `SafeNativeError::InvariantViolation(UNKNOWN_INVARIANT_VIOLATION_ERROR)` when there aren't
+/// enough arguments on the stack.
+#[macro_export]
+macro_rules! safely_pop_arg {
+    ($args:ident, $t:ty) => {{
+        use $crate::reexports::move_vm_types::natives::function::{PartialVMError, StatusCode};
+        match $args.pop_back() {
+            Some(val) => match val.value_as::<$t>() {
+                Ok(v) => v,
+                Err(e) => return Err($crate::SafeNativeError::InvariantViolation(e)),
+            },
+            None => {
+                return Err($crate::SafeNativeError::InvariantViolation(
+                    PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR),
```

Comment:
this seems like nightmare to debug if it happens

Reply:
Thanks to @gerben-stavenga's recent work, we now include a stack trace as part of the invariant violation error, so I wouldn't worry too much about debuggability for now.

Fully agreed this API is terrible though. It was created because Move VM didn't offer good support in hte first place. The plan is to replace it with some proper user-friendly APIs in the future.